server/jamovi/server/formatio/ods.py

```diff
@@ -1,21 +1,64 @@
 
-from ezodf import opendoc
+from math import isnan as float_isnan
+from itertools import islice
+from ezodf import opendoc, newdoc, Table
 
-from .reader import Reader
+from jamovi.core import DataType
+from jamovi.server.instancemodel import InstanceModel
 
-from itertools import islice
+from .reader import Reader
 
 
 def get_readers():
     return [ ( 'ods', read ) ]
 
 
+def get_writers():
+    return [ ( 'ods', write ) ]
+
+
 def read(data, path, prog_cb, *, settings, **kwargs):
 
     reader = ODSReader(settings)
     reader.read_into(data, path, prog_cb)
 
 
+def write(data: InstanceModel, path, prog_cb):
+
+    spreadsheet = newdoc(doctype = "ods", filename = path)
+
+    def should_exclude(column):
+        return column.is_virtual or (column.is_filter and column.active)
+
+    def get_valtype(data_type):
+        return 'float' if data_type in [DataType.DECIMAL, DataType.INTEGER] else 'string'
+
+    def isnan(value):
+        return (isinstance(value, int) and value == -2147483648) or (isinstance(value, float) and float_isnan(value))
+
+    cols = [ col for col in data if not should_exclude(col) ]
+    col_no = [ col.index for col in cols ]
+    col_type = [ get_valtype(col.data_type) for col in cols ]
+
+    ws = spreadsheet.sheets.append(Table('Sheet 1', size = (data.row_count + 1, len(cols))))
```

Comment:
you can push this guy across multiple lines too ... but you'll want to use `data.row_count_ex_filtered` (which gives you the row count when filtered out rows are excluded).